src/image.py

```diff
@@ -70,6 +73,22 @@ def is_child_image(self):
     def is_test_promotion_enabled(self):
         return bool(self.info.get('enable_test_promotion'))
 
+    @property
+    def corresponding_common_stage_image(self):
+        """
+        Retrieve the corresponding common stage image for a given image.
+        """
+        return self._corresponding_common_stage_image
+
+    @corresponding_common_stage_image.setter
+    def corresponding_common_stage_image(self, docker_image_object):
+        """
+        Sets the value for the corresponding_common_stage_image variable.
+        """
+        if self.to_push:
+            raise ValueError("Corresponding common stage image can only exist if the image is non-pushable")
```

Comment:
Why can this only exist if the image is non-pushable?

Reply:
Changes as per conversation.